tests/collections/asr/decoding/test_batched_rnnt_decoding.py

```diff
@@ -0,0 +1,840 @@
+    logging.info(f"  Total: {mem.total / (1024 ** 3):.2f} GB")
+    logging.info(f"  Used: {mem.used / (1024 ** 3):.2f} GB")
+    logging.info(f"  Free: {mem.available / (1024 ** 3):.2f} GB")
+    logging.info(f"  Percent Used: {mem.percent}%")
+
+    logging.info("\nSwap Usage:")
+    logging.info(f"  Total: {swap.total / (1024 ** 3):.2f} GB")
+    logging.info(f"  Used: {swap.used / (1024 ** 3):.2f} GB")
+    logging.info(f"  Free: {swap.free / (1024 ** 3):.2f} GB")
+    logging.info(f"  Percent Used: {swap.percent}%")
+
+
+faulthandler.enable()
+
+RNNT_MODEL = "stt_en_conformer_transducer_small"
+TDT_MODEL = "nvidia/parakeet-tdt_ctc-110m"
+DEVICES = [torch.device("cpu")]
+
+if torch.cuda.is_available():
+    DEVICES.append(torch.device('cuda'))
+
+NUMBA_RNNT_LOSS_AVAILABLE = numba_utils.numba_cpu_is_supported(
+    __NUMBA_MINIMUM_VERSION__
+) or numba_utils.numba_cuda_is_supported(__NUMBA_MINIMUM_VERSION__)
+
+
+@pytest.fixture()
+def test_audio_filenames(test_data_dir):
+    return tuple(glob.glob(os.path.join(test_data_dir, "asr", "test", "an4", "wav", "*.wav")))
+
+
+@lru_cache(maxsize=4)
+def get_model(model_name: str, device: torch.device = torch.device("cpu")):
+    return ASRModel.from_pretrained(model_name, map_location=device)
+
+
+@lru_cache(maxsize=4)
+def get_model_encoder_output(
+    test_audio_filenames,
+    num_samples: int,
+    model_name: ASRModel,
+    device: torch.device = torch.device("cpu"),
+    dtype: torch.dtype = torch.float32,
+):
+    audio_filepaths = test_audio_filenames[:num_samples]
+
+    get_memory_info("Before dataset loading")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with open(os.path.join(tmpdir, 'manifest.json'), 'w', encoding='utf-8') as fp:
+            for audio_file in audio_filepaths:
+                entry = {'audio_filepath': audio_file, 'duration': None, 'text': ''}
+                fp.write(json.dumps(entry) + '\n')
+
+        config = {
+            'paths2audio_files': audio_filepaths,
+            'batch_size': len(audio_filepaths),
+            'temp_dir': tmpdir,
+            'num_workers': 1,
+        }
+
+        with torch.no_grad():
+            model = get_model(model_name, device)
+            model.preprocessor.featurizer.dither = 0.0
+            model.preprocessor.featurizer.pad_to = 0
+            model.eval()
+
+            temporary_datalayer = model._setup_transcribe_dataloader(config)
+            get_memory_info("Before loop")
+            for test_batch in temporary_datalayer:
+                encoded, encoded_len = model.forward(
+                    input_signal=test_batch[0].to(device, dtype=dtype), input_signal_length=test_batch[1].to(device)
+                )
+            get_memory_info("After dataset loading")
+    logging.info("Successfully loaded dataset")
+    return model, encoded, encoded_len
+
+
+def decode_text_from_hypotheses(hyps, decoding):
+    decoded_hyps = decoding.decode_hypothesis(hyps)  # type: List[str]
```

Comment:
Please avoid python2-style type annotations.
E.g., here should be `decoded_hyps: list[str] = ...`